waku/v2/protocol/rln/group_manager/dynamic/membership_fetcher_test.go

```diff
@@ -0,0 +1,77 @@
+package dynamic
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/waku-org/go-waku/waku/v2/protocol/rln/contracts"
+	"github.com/waku-org/go-waku/waku/v2/protocol/rln/group_manager"
+	"github.com/waku-org/go-waku/waku/v2/protocol/rln/web3"
+	"github.com/waku-org/go-waku/waku/v2/utils"
+	"github.com/waku-org/go-zerokit-rln/rln"
+)
+
+var NULL_ADDR common.Address = common.HexToAddress("0x0000000000000000000000000000000000000000")
+
+func TestFetchingLogic(t *testing.T) {
+	client := NewMockClient(t, "membership_fetcher.json")
+
+	rlnContract, err := contracts.NewRLN(NULL_ADDR, client)
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
Use this to not have to check for nil, and have the testify do it for you. (Same suggestion applies to line 31, 56, etc
```suggestion
	require.NoError(t, err)
```